src/components/ae-input/ae-input.vue

```diff
@@ -9,14 +9,19 @@
         <slot name="header" />
       </div>
       <!-- Input tag -->
-      <input :type="type"
-             :id="id"
-             :placeholder="placeholder"
-             class="ae-input"
-             :class="{ aemount }"
-             @focus="focus = true"
-             @blur="focus = false"
-             @input="propagate" />
+      <input
+        v-if="!$slots.default"
+        :id="id"
+        :value="value"
+        :type="type"
+        :placeholder="placeholder"
+        class="ae-input"
+        :class="{ aemount }"
+        @focus="focus = true"
+        @blur="focus = false"
+        @input="propagateValue"
+      />
+      <slot />
```

Comment:
How this is supposed to be used? Maybe we should make additional components instead? Passing anything to default slot shouldn't break properties of this component (`value`, `placeholder`, `focus` and other fields becomes useless).

Reply:
@davidyuk 

In this case is a functionality that we want! Because providing that flexibility means I can add different type of inputs / content in there and just inherit the styles and some functionality from the `ae-input` component.

There are some edge cases that this supports. And it actually does not break anything, since the values just don't get passed to the `slot` component.

Reply:
I don't agree if you want to have a wrapper around an arbitrary input so extract a separate component that will not have `value`, `placeholder`, `focus` properties. And use this component in AeInput and any other similar components.

Reply:
@davidyuk 

A workaround to accessing the `parent` context is to use `scoped-slots`, in this way we can access anything passed to the parent element that has to be used inside the child slot.

This slot is for flexibility, while working with token-migration, there were cases where we needed a slightly different `<input />` element, and in my opinion having multiple `<input />` components is not that good of an idea.

Using `<slot />` will only be useful in edge cases, so normally the user would simply want to use the styles we've provided but in these edge cases the `<slot />` gives a better solution

Reply:
> and in my opinion having multiple <input /> components is not that good of an idea

Six months or so developers of components package decided to don't mix different types of inputs in a single component. I think still we should work according to it.